### PR TITLE
Testing n+1

### PR DESCRIPTION
**Pull request by anas599**
In this PR:

- Fixed n+1 query issue for improved performance.
- Added Capybara tests to enhance test coverage.
- Included RSpec Capybara tests for User and Post index/show pages.

Screenshot 2023-05-15 at 6 13 57 PM
Screenshot 2023-05-15 at 6 24 50 PM

### CODE REVIEW

File: test/system/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      expect(page).to have_content('image')
+    end
+
+    it 'should navigate to the user show page' do
+      click_link('Tom1', exact_text: true)
+      sleep 1
+      expect(current_path).to eq(user_path(user.id))
+    end
+  end
+  describe 'Test User show' do
```
Comment (brytebee) [inline test/system/user_spec.rb:43]: The tests for the user show page should include the following: 
- the user's bio
- the user's first 3 posts
- the redirects to the post's show page when you click on a user's post.
- that redirects to the post's show page when you click on a user's post.
- that redirects to the post's show page when you click on a user's post.

image

File: test/system/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
+require 'rails_helper'
+
+RSpec.describe 'Post index/show', type: :system do
+  describe 'post index page' do
```
Comment (brytebee) [inline test/system/post_spec.rb:4]: The Post index/show tests are missing the following:
- a test for the user image
- the no. of posts made
- how many comments are on the page
- how many likes are on the page

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @anas599 & @hopeforkoami & @V0pe 👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: test/system/post_spec.rb (hunk 2 above)
Comment (omar-muhamad) [reply test/system/post_spec.rb:4]: - [ ] As requested by the last reviewer, Please add those tests to the Post index/show views:

- a test for the user image
- the no. of posts made
- how many comments are on the page
- how many likes are on the page

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @anas599,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Davidosky007 — APPROVED** [review]
Hi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 
* All test cases covered 
* N+1 problem solved 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by addod19** [comment]
### Hi TEAM, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because you submitted a project that is already approved and merged

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by anas599** [comment]
> ### Hi TEAM,
> While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
> 
> ## Invalid Code Review Request
> You have submitted a project, that is invalid because you submitted a project that is already approved and merged
> 
> Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _Invalid Code Review Request **does not count** into the code reviews limit._

**Hi @addod19 the pull request was approved you can check it above and I closed the pull request because it got approved
https://github.com/anas599/blog/pull/9#pullrequestreview-1429174908**

**Pull request merged**